x-pack/plugins/apm/server/lib/helpers/transaction_query.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import {
+  TRANSACTION_DURATION,
+  TRANSACTION_NAME
+} from '../../../common/constants';
+import { ITransactionGroup } from '../../../typings/TransactionGroup';
+
+interface ITransactionGroupSample {
+  transaction: {
+    id: string;
+    type: string;
+  };
+  context: {
+    service: {
+      name: string;
+    };
+  };
+}
```

Comment:
We should extract transaction and context from [here](https://github.com/elastic/kibana/blob/apm/dt/x-pack/plugins/apm/typings/Transaction.ts) and then do something like:

```
interface ITransactionGroupSample {
  transaction: TransactionTransaction;
  context: TransactionContext;
}
```

Reply:
Btw. I'm very open to better names than `TransactionTransaction ` :D

Reply:
... even `TransactionDocTransaction` is better but that would require us to rename the `Transaction` interface to `TransactionDoc`